TESTS/mbed_drivers/lp_timer/main.cpp

```diff
@@ -0,0 +1,410 @@
+     * the timer is stopped does not have influence on the
+     * timer counted time. */
+    wait_ms(50);
+
+    /* ------ */
+
+    /* Start the timer. */
+    lp_timer.start();
+
+    /* Wait 20 ms. */
+    wait_ms(20);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - totally 30 ms have elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(2 * DELTA_S, 0.030f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 30, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(2 * DELTA_US, 30000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(2 * DELTA_US, 30000, lp_timer.read_high_resolution_us());
+
+    /* Wait 50 ms - this is done to show that time elapsed when
+     * the timer is stopped does not have influence on the
+     * timer counted time. */
+
+    /* ------ */
+
+    /* Start the timer. */
+    lp_timer.start();
+
+    /* Wait 30 ms. */
+    wait_ms(30);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - totally 60 ms have elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(3 * DELTA_S, 0.060f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 60, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(3 * DELTA_US, 60000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(3 * DELTA_US, 60000, lp_timer.read_high_resolution_us());
+
+    /* Wait 50 ms - this is done to show that time elapsed when
+     * the timer is stopped does not have influence on the
+     * timer time. */
+    wait_ms(50);
+
+    /* ------ */
+
+    /* Start the timer. */
+    lp_timer.start();
+
+    /* Wait 1 sec. */
+    wait_ms(1000);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - totally 5060 ms have elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(4 * DELTA_S, 1.060f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 1060, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(4 * DELTA_US, 1060000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(4 * DELTA_US, 1060000, lp_timer.read_high_resolution_us());
+}
+
+/* This test verifies if reset() function resets the
+ * low power timer counted time.
+ *
+ * Timer has been started and stopped once, then reset
+ * operation was performed.
+ * When timer is started and stopped next time.
+ * Then timer read functions returns only the the second
+ * measured time.
+ */
+void test_lptimer_reset()
+{
+    LowPowerTimer lp_timer;
+
+    /* First measure 10 ms delay. */
+    lp_timer.start();
+
+    /* Wait 10 ms. */
+    wait_ms(10);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - totally 10 ms elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(DELTA_S, 0.010f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 10, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(DELTA_US, 10000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(DELTA_US, 10000, lp_timer.read_high_resolution_us());
+
+    /* Reset the timer - previous measured time should be lost now. */
+    lp_timer.reset();
```

Comment:
Shall we validate the reset here? Rather than start the timer again?

Reply:
The intention of this test case is to show that reset() function works - i.e. clears time already counted by the timer. Verification if timer counts time between starts and stops is provided in test_lptimer_time_accumulation test function.